app/src/main/java/org/fossasia/pslab/communication/ScienceLab.java

```diff
@@ -1153,6 +1151,800 @@ public int calculateDigitalChannel(String name) {
         }
     }
 
+    private Double getHighFrequencyBackup(String pin) {
+        try {
+            mPacketHandler.sendByte(mCommandsProto.COMMON);
+            mPacketHandler.sendByte(mCommandsProto.GET_HIGH_FREQUENCY);
+            mPacketHandler.sendByte(this.calculateDigitalChannel(pin));
+            int scale = mPacketHandler.getByte();
+            long value = mPacketHandler.getLong();
+            mPacketHandler.getAcknowledgement();
+            return scale * value / 1.0e-1;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public Double getHighFrequency(String pin) {
+        /*
+        Retrieves the frequency of the signal connected to ID1. for frequencies > 1MHz
+		also good for lower frequencies, but avoid using it since
+		the oscilloscope cannot be used simultaneously due to hardware limitations.
+		The input frequency is fed to a 32 bit counter for a period of 100mS.
+		The value of the counter at the end of 100mS is used to calculate the frequency.
+        */
+        try {
+            mPacketHandler.sendByte(mCommandsProto.COMMON);
+            mPacketHandler.sendByte(mCommandsProto.GET_ALTERNATE_HIGH_FREQUENCY);
+            mPacketHandler.sendByte(this.calculateDigitalChannel(pin));
+            int scale = mPacketHandler.getByte();
+            long value = mPacketHandler.getLong();
+            mPacketHandler.getAcknowledgement();
+            return scale * value / 1.0e-1;
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public double getFrequency(String channel, Integer timeout) {
```

Comment:
Since every other functions in the vicinity of this function is using reference type `Double` would it be uniform if we can use the same data type here other than primitive `double`? Just a thought! :blush: